drivers/entropy/entropy_npcx_drbg.c

```diff
@@ -0,0 +1,246 @@
+	enum ncl_status (*clear)(void *ctx);
+};
+#define NPCX_NCL_DRBG ((const struct npcx_ncl_drbg *)NPCX_NCL_DRBG_BASE_ADDR)
+
+/* The 2nd index of the reg property holds the address of NCL_SHA_Power ROM API */
+#define NPCX_NCL_SHA_POWER_ADDR ((const struct npcx_ncl_drbg *)DT_INST_REG_ADDR_BY_IDX(0, 1))
+struct npcx_ncl_sha {
+	/* Power on/off SHA module. */
+	enum ncl_status (*power)(void *ctx, uint8_t on);
+};
+#define NPCX_NCL_SHA_POWER ((const struct npcx_ncl_sha *)NPCX_NCL_SHA_POWER_ADDR)
+
+static int entropy_npcx_drbg_enable_sha_power(void *ctx, bool enable)
+{
+	enum ncl_status ncl_ret;
+
+	ncl_ret = NPCX_NCL_SHA_POWER->power(ctx, enable);
+	if (ncl_ret != NCL_STATUS_OK) {
+		LOG_ERR("Fail to %s SHA power: err 0x%02x", enable ? "enable" : "disable", ncl_ret);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int entropy_npcx_drbg_enable_drbg_power(void *ctx, bool enable)
+{
+	enum ncl_status ncl_ret;
+
+	ncl_ret = NPCX_NCL_DRBG->power(ctx, enable);
+	if (ncl_ret != NCL_STATUS_OK) {
+		LOG_ERR("Fail to %s DRBG power: err 0x%02x", enable ? "enable" : "disable",
+			ncl_ret);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int entropy_npcx_drbg_suspend(const struct device *dev)
+{
+	struct entropy_npcx_drbg_dev_data *const data = dev->data;
+	void *ctx = data->handle;
+
+	return entropy_npcx_drbg_enable_drbg_power(ctx, false);
+}
+
+static int entropy_npcx_drbg_resume(const struct device *dev)
+{
+	struct entropy_npcx_drbg_dev_data *const data = dev->data;
+	void *ctx = data->handle;
+
+	return entropy_npcx_drbg_enable_drbg_power(ctx, true);
+}
+#endif /* CONFIG_PM_DEVICE */
+
+static int entropy_npcx_drbg_get_entropy(const struct device *dev, uint8_t *buf, uint16_t len)
+{
+	struct entropy_npcx_drbg_dev_data *const data = dev->data;
+	enum ncl_status ncl_ret;
+	void *ctx = data->handle;
+	int ret = 0;
+
+	k_sem_take(&data->sem_lock, K_FOREVER);
+
+	ret = entropy_npcx_drbg_enable_sha_power(ctx, true);
+	if (ret != 0) {
+		goto err_exit;
+	}
+
+	ncl_ret = NPCX_NCL_DRBG->generate(ctx, NULL, 0, buf, len);
+	if (ncl_ret != NCL_STATUS_OK) {
+		LOG_ERR("Fail to generate: err 0x%02x", ncl_ret);
+		ret = -EIO;
+		goto err_exit;
+	}
+
+	ret = entropy_npcx_drbg_enable_sha_power(ctx, false);
+	if (ret != 0) {
+		goto err_exit;
+	}
+
+err_exit:
+	k_sem_give(&data->sem_lock);
+
+	return ret;
+}
+
+static int entropy_npcx_drbg_init(const struct device *dev)
+{
+	struct entropy_npcx_drbg_dev_data *const data = dev->data;
+	uint32_t handle_size_required;
+	enum ncl_status ncl_ret;
+	void *ctx = data->handle;
+	int ret;
+
+	handle_size_required = NPCX_NCL_DRBG->get_context_size();
+	if (handle_size_required != NPCX_DRBG_HANDLE_SIZE) {
+		LOG_ERR("Unexpected NCL DRBG context_size = %d", handle_size_required);
+		return -ENOSR;
+	}
+
+	ret = entropy_npcx_drbg_enable_sha_power(ctx, true);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = entropy_npcx_drbg_enable_drbg_power(ctx, true);
+	ncl_ret = NPCX_NCL_DRBG->power(ctx, true);
```

Comment:
stale? this is called by `entropy_npcx_drbg_enable_drbg_power`

Reply:
Nice catch! I forgot to remove it when I refactor the drbg power enable/disable.